pkg/audit/audit_cache_lister.go

```diff
@@ -25,14 +24,19 @@ type CacheLister struct {
 	// Caution: only to be read from while watched is locked, such as through
 	// DoForEach.
 	auditCache client.Reader
-	// watched is the set of objects watched by the audit cache.
-	watched *watch.Set
+	// lister is a delegate like cachemanager that we can use to query a watched set of GKVs.
```

Comment:
nit: what is a "delegate"?  

I think it would be more helpful to write a comment about why we do this.

"Passing our logic as a callback to a watched.Set allows us to take actions while holding the lock on the watched.Set.  This prevents us from querying the API server for kinds that aren't currently being watched by the CacheManager."

Reply:
i think the `delegate` part is pretty well understood after including your suggested comment 💯